cmd/symbols/proto/conversion.go

```
@@ -0,0 +1,341 @@
+package proto
+
+import (
+	"github.com/sourcegraph/sourcegraph/internal/api"
+	"github.com/sourcegraph/sourcegraph/internal/search"
+	"github.com/sourcegraph/sourcegraph/internal/search/result"
+	"github.com/sourcegraph/sourcegraph/internal/types"
+	"google.golang.org/protobuf/types/known/durationpb"
+)
+
+func (x *SearchRequest) FromInternal(p *search.SymbolsParameters) {
+	if x == nil || p == nil {
+		return
+	}
```

IMO, panicking here is better behavior than silently doing nothing. I don't think we want to be quite this defensive because we'll end up quietly doing the wrong thing. In particular, `x == nil` means that someone used this method incorrectly.

---

Alright, I'll tweak everything besides the SymbolInfoResponse's FromInternal method. For that one, 

It's quite common that that implementation returns a `nil`  *types.SymbolInfo: 

https://github.com/sourcegraph/sourcegraph/blob/9cc0f5e7f26ec3807875804c54915b745f1e7a5c/cmd/symbols/squirrel/service.go#L55-L125

We'll need to explicitly expect `nil` in this case.